### PR TITLE
Add basic Ruby-to-Swift migration

### PR DESCRIPTION
**Pull request by getaaron**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation, Context, and Description
For @taquitos to review (and anyone else interested)

For the Fastfile-in-Swift feature, this change:

1. Gives a helpful error message if users run `fastlane init --swift` when they already have a Ruby fastlane project
2. Offers to rename their fastlane directory to `fastlane.old` and create a new Swift-based setup if they so desire

### CODE REVIEW

File: fastlane/lib/fastlane/commands_generator.rb [hunk 1]
```diff
@@ -121,7 +121,6 @@ def run
           else
             is_swift_fastfile = args.include?("swift")
             Fastlane::Setup.new.run(user: options.user, is_swift_fastfile: is_swift_fastfile)
-            Fastlane::SwiftLaneManager.first_time_setup
```
Comment (getaaron) [inline fastlane/lib/fastlane/commands_generator.rb]: I moved these calls into `Fastlane::Setup` to (1) reduce duplication, and (2) avoid a crash if `Fastlane::Setup` returns when a project is already configured, but then `first_time_setup` runs anyway

**Comment by getaaron** [comment]
Circle is failing for some leftover pry comments (unrelated to my change) but tests are passing locally

File: fastlane/lib/fastlane/setup/setup.rb [hunk 2]
```diff
@@ -27,6 +42,7 @@ def run(user: nil, is_swift_fastfile: false)
 
       if platform == :ios
         SetupIos.new.run(user: user, is_swift_fastfile: is_swift_fastfile)
+        Fastlane::SwiftLaneManager.first_time_setup if is_swift_fastfile
```
Comment (getaaron) [inline fastlane/lib/fastlane/setup/setup.rb]: This should maybe be moved even deeper (into the `SetupIos` class)

**Comment by getaaron** [comment]
@taquitos Any thoughts on this?

File: fastlane/lib/fastlane/setup/setup.rb (hunk 2 above)
Comment (taquitos) [reply fastlane/lib/fastlane/setup/setup.rb]: I agree, and it could run at the end of `SetupIos.new.run`

**Comment by taquitos** [comment]
This looks awesome, just 1 comment which you already pointed out. Let me know what you think

**Comment by taquitos** [comment]
Oh yeah, and `rubocop -a`

**Comment by janpio** [comment]
Hey @getaaron, a rebase onto current master to get current CI setup and what @taquitos said to fix the code style stuff would be nice. Thank you.

**Comment by getaaron** [comment]
@janpio Done. I ran `rubocop -a` locally and had to extract some code into a separate `self.already_set_up` method

File: fastlane/lib/fastlane/setup/setup.rb [hunk 3]
```diff
@@ -123,6 +115,33 @@ def self.start(user: nil, is_swift_fastfile: false)
     end
     # rubocop:enable Metrics/BlockNesting
 
+    def self.already_set_up(user: nil, is_swift_fastfile: false)
+      if is_swift_fastfile && !FastlaneCore::FastlaneFolder.swift?
+        UI.important("fastlane is already configured using Ruby.")
+        UI.message("If you want to switch to Swift, I can reinitialize your fastlane project:")
+        UI.message(" 1️⃣  The fastlane directory (at #{FastlaneCore::FastlaneFolder.path}) would be renamed `fastlane.old`")
+        UI.message(" 2️⃣  I would run the fastlane Swift initialization")
+        UI.message(" 3️⃣  You'd need to port your customizations from Ruby to Swift by hand.")
```
Comment (janpio) [inline fastlane/lib/fastlane/setup/setup.rb]: what "customizations" does this refer to? my own lanes in the Fastfile?

File: fastlane_core/lib/fastlane_core/fastlane_folder.rb [hunk 4]
```diff
@@ -67,5 +67,12 @@ def self.create_folder!(path = nil)
       FileUtils.mkdir_p(path)
       UI.success("Created new folder '#{path}'.")
     end
+
+    def self.append_old!
```
Comment (janpio) [inline fastlane_core/lib/fastlane_core/fastlane_folder.rb]: why `append_old`? What does this mean in this context?

File: fastlane/lib/fastlane/setup/setup.rb (hunk 3 above)
Comment (getaaron) [reply fastlane/lib/fastlane/setup/setup.rb]: Yeah, anything in a Fastfile, Matchfile, Appfile, etc. which would typically be parsed as Ruby.

File: fastlane/lib/fastlane/setup/setup.rb [hunk 5]
```diff
@@ -123,6 +115,33 @@ def self.start(user: nil, is_swift_fastfile: false)
...
+        if UI.confirm("Should I rename your folder \"fastlane.old\" and initialize a new Swift fastlane project?")
```
[the remark this answers is not in the document]
Comment (getaaron) [reply fastlane/lib/fastlane/setup/setup.rb]: Sometimes it's called `.fastlane` (hidden) which is why I decided to print the path in step 1

[the remark this answers is not in the document]
Comment (janpio) [reply fastlane/lib/fastlane/setup/setup.rb]: Ok, then only the "to" is missing.

File: fastlane_core/lib/fastlane_core/fastlane_folder.rb (hunk 4 above)
Comment (janpio) [reply fastlane_core/lib/fastlane_core/fastlane_folder.rb]: Ok, that was _too_ obvious.

File: fastlane/lib/fastlane/setup/setup.rb (hunk 3 above)
Comment (janpio) [reply fastlane/lib/fastlane/setup/setup.rb]: Can we somehow formulate this in a way that makes it clearer? I though of customizations in fastlane first - which of course no one really does. Or is this just me?

**Comment by joshdholtz** [comment]
@getaaron Are you still interested in trying to get this one merged in? It looks like we got some conflicts 🙃

**Pull request closed without merging**